modules/org.restlet.ext.apispark/src/org/restlet/ext/apispark/internal/conversion/swagger/v2_0/Swagger2Translator.java

```diff
@@ -581,4 +617,523 @@ public static Swagger getSwagger(Definition definition) {
+        });
+
+        model.setProperties(new LinkedHashMap<String, Property>());
+        for (Iterator<Map.Entry<String, Property>> it = propertyList.iterator(); it.hasNext();) {
+            Map.Entry<String, Property> entry = it.next();
+            model.getProperties().put(entry.getKey(), entry.getValue());
+        }
+    }
+
+    /**
+     * Removes the dots in maximums and minimums for integer and long properties.
+     * 
+     * @param swaggerProperty
+     *            The Swagger property to process.
+     * @param property
+     *            The RWADef property to update
+     */
+    private static void formatPropertyMinMax(AbstractNumericProperty swaggerProperty,
+            org.restlet.ext.apispark.internal.model.Property property) {
+        String min = SwaggerUtils.toString(swaggerProperty.getMinimum());
+        String max = SwaggerUtils.toString(swaggerProperty.getMaximum());
+
+        if (swaggerProperty instanceof IntegerProperty
+                || swaggerProperty instanceof LongProperty) {
+            property.setMin(min == null ? null : min.replaceAll("\\.[0-9]+", ""));
+            property.setMax(max == null ? null : max.replaceAll("\\.[0-9]+", ""));
+
+        } else {
+            property.setMin(min);
+            property.setMax(max);
+
+        }
+    }
+
+    private static void fillResources(Swagger swagger, Contract contract,
+            List<String> produces, List<String> consumes,
+            Map<String, org.restlet.ext.apispark.internal.model.Parameter> parameters) {
+        if (swagger.getPaths() == null) {
+            return;
+        }
+
+        for (String key : swagger.getPaths().keySet()) {
+            Resource resource = new Resource();
+            Path path = swagger.getPath(key);
+
+            // TODO: description not implemented in Swagger 2.0
+            resource.setName(ConversionUtils.processResourceName(key));
+            resource.setResourcePath(key);
+            for (Operation operation : path.getOperations()) {
+                resource.addSections(operation.getTags());
+            }
+
+            if (path.getParameters() != null) {
+                for (Parameter parameter : path.getParameters()) {
+                    PathVariable pathVariable = null;
+                    if (parameter instanceof PathParameter) {
+                        pathVariable = new PathVariable();
+                        fillRwadefPathVariable(pathVariable, (PathParameter) parameter);
+                    } else if (parameter instanceof RefParameter) {
+                        RefParameter refParameter = (RefParameter) parameter;
+                        org.restlet.ext.apispark.internal.model.Parameter savedParameter =
+                                parameters.get(refParameter.getSimpleRef());
+                        if (savedParameter instanceof PathVariable) {
+                            pathVariable = (PathVariable) savedParameter;
+                        }
+                    }
```

Comment:
There should not be any other type so I'll add the else and log (warn) but we should continue.